### ISSUE TITLE
ShouldBe(True|False|Null)

### ISSUE THREAD

**Issue by JakeGinnivan** [shouldly/shouldly#248]
From a NuGet contact us request, it was suggested we add ShouldBe(True|False|Null) as convenience methods.

What are peoples thoughts, it would mean:

```
true.ShouldBe(false) -> true.ShouldBeFalse();
true.ShouldNotBe(true) -> true.ShouldNotBeTrue();
someNullVariable.ShouldBe(null) -> someNullValue.ShouldBeNull()
```

Would people find these useful, does it improve the readability?

My initial reaction is I like the null one, not so keen on the true/false ones.

**Comment by Moily** [comment]
My thoughts, as the original requestor for this... (thanks for pointing me here, Jake)

The requirement for these useful shortcuts comes from using similar constructs in production code. Once you start using .ShouldBeTrue(), .ShouldBeFalse, .ShouldBeNull(), .ShouldNotBeNull() it does slim down the assertion code and improve readability, imo.

I'm not convinced we would need .ShouldNotBeTrue() and .ShouldNotBeFalse() as these would be redundant due to the first two, above, but I would see why they could be included for completeness-sakes.

**Comment by JakeGinnivan** [comment]
If we separate this into two different items

1) Add `ShouldBeNull()` and `ShouldNotBeNull()`
2) Add `ShouldBeTrue()` and `ShouldBeFalse()`

I vote for lets add 1, lets not add 2. The thing is, all these examples are the same number of characters.

If you submit a PR for the null overloads I am happy to merge. I would like more feedback from other users for 2 first

**Comment by JoeMighty** [comment]
I kind of like the idea of ShouldBe(true|fase) and can see the value in it (changing value at runtime for test cases for instance). Out of interest, what is your reasoning against 2 that does not apply to 1?

**Comment by Moily** [comment]
You are right in that it's the same number of characters, so we're not actually slimming down the code as such. Perhaps the readability is down to what we've been used to seeing, i.e

`someVar.ShouldBeNull();`

seems (subjectively) more correct than

`someVar.ShouldBe(null);`

in that null is usually a specialised state over a given instance to compare against. 

Likewise .ShouldNotBeNull() is a good catch before inspecting discrete properties of the object being inspected so a test failure would identify that the object wasn't set at all rather than the first call to access a member failing due to the object being null.

Again, both approaches are subjective so I can see why you wouldn't be for their inclusion.

**Comment by JakeGinnivan** [comment]
The reason I am happy for the null case is I don't see null as a value. It's a state, or a lack of a value.

On the other hand, true/false are values and I see them just like I see a string value or some other value?

**Comment by JakeGinnivan** [comment]
That is my intiial gut reaction. I am not against us adding it if the majority thing the ShouldBeTrue/False add value

**Comment by Moily** [comment]
Agreed that the bool values are just values but again they can be seen as states, hence the usefulness of having strongly-typed methods for them rather than relying on the instance type of the argument being passed into .ShouldBe()

Also it enables Code Completion to save typing out the full .ShouldBe(true|false) !

**Comment by JakeGinnivan** [comment]
Are you able to submit a pull request? :)

**Comment by Moily** [comment]
Sure, I can do one for each case (your #1 and #2), if you want to decide which to take at a later date?

**Comment by JakeGinnivan** [comment]
Sounds good. Having a quick look most other frameworks have true/false methods. So if you submit both separately I will merge both

**Comment by JoeMighty** [comment]
How are you getting on with this one @Moily? I've got some spare time and happy to pick some of this up if you like?

**Comment by Moily** [comment]
Hi Joe,

Apologies but work and a tricky house move has meant I won't be able to get
round to this for a few weeks or so.

Happy to let you pick up anything you like, if you're ok with that.

Regards,

Rhys

**Comment by Moily** [comment]
Finding spare time is tricky at the moment but I can look at it in a couple of weeks, unless anyone else wants to get there first?

**Comment by JoeMighty** [comment]
I'm happy to take a look. I saw you removed your original post so presumed you'd found some time to work on it :)

**Comment by JoeMighty** [comment]
ShouldBeTrue/False extensions have been committed (#258)

**Comment by JakeGinnivan** [comment]
Closed by #258
